app/assets/javascripts/foreman_docker/container_image_search.js

```diff
@@ -0,0 +1,203 @@
+function ContainerImageSearch() {
+  this.initialize = function (registryType) {
+    this.registryType = registryType;
+    this.form = $('form[data-registry="' + this.registryType + '"]');
+    this.inputs = {
+      image: this.getInput('image'),
+      tag: this.getInput('tag')
+    };
+    this.results = this.form.find('.registry-search-results')
+    this.resultsList = this.results.find('.registry-search-results-list');
+    this.resultsList.on('click', this.selectImage.bind(this));
+
+    this.searchButton = this.form.find('.image-search-button');
+    this.searchButton.on('click', function (event) {
+      event.preventDefault();
+      this.fullResultList();
+    }.bind(this))
+
+    this.setupInputs();
+  };
+
+  this.registryId = function () {
+    return $('#docker_container_wizard_states_image_registry_id').val();
+  };
+
+  this.getInput = function (input) {
+    return this.form.find('input[data-' + input + ']:first');
+  };
+
+  this.getFormGroup = function (input) {
+    return input.closest('.form-group');
+  }
+
+  this.getSpinner = function (input) {
+    return this.getFormGroup(input).find('.autocomplete-status');
+  };
+
+  this.getInlineHelp = function (input) {
+    return this.getFormGroup(input).find('.help-inline');
+  }
+
+  this.validRequest = function () {
+    return (this.registryType == 'registry' && this.registryId() != '' ||
+           this.registryType == 'hub' && this.registryId() == '') &&
+           this.inputs.image.val() != '';
+  }
+
+  this.getAutocompleteResults = function (tag, input, callback, params) {
+    if(!this.validRequest())
+      return;
+
+    var spinner = this.getSpinner(tag),
+        imageName = this.inputs.image.val(),
+        tagsOnly = tag.data('tag'),
+        params = $.extend({
+          registry: this.registryType,
+          search: tagsOnly ? imageName + ':' + input.term : input.term,
+          registry_id: this.registryId(),
+          tags: tagsOnly
+        }, params)
+
+    spinner.removeClass('pficon pficon-error-circle-o pficon-ok')
+           .addClass('spinner spinner-xs').show();
+
+    $.getJSON(tag.data("url"), params, function (data) {
+      this.setAutocompleteConfirmationStatus(tag, data)
+      callback(data);
+    }.bind(this))
+    .error(function(result) {
+      $.jnotify(result.responseText, 'error', true);
```

Comment:
please use the notifify function instead of calling jnotify directly, as it will be deprecated.